### ISSUE TITLE
Test262/eshost runtime feature request: -module flag for ch 

### ISSUE THREAD

**Issue by rwaldron** [Microsoft/ChakraCore#5274]
TL;DR:

`eshost` needs a way to invoke `ch` with a flag that instructs `ch` to treat the `<source file>` as module code. e.g. `ch -module source.js`

All other ECMAScript hosts presently expose a flag that enables this "mode":

| Engine | Host | Existing Mechanism |
| ---- | ---- | ---------------------|
| ChakraCore | ch  | Must wrap source in a call to `WScript.LoadModule(code)` or `WScript.LoadScriptFile(file, 'module')` \* |
| JavaScriptCore | jsc | `jsc -m source.js` |
| SpiderMonkey | jsshell | `js --module source.js` |
| V8 | d8 | `d8 --module source.js` |
| XS | xs | `xs -m source.js` |

**Comment by rhuanjl** [comment]
If wanted by the team I could make a PR to implement this. Please let me know if you’d like me to.

**Comment by jackhorton** [comment]
I had looked into making this change in the past because I wanted Intl tests to be modules. The logic in `ch` is far too complicated right now, so it diverged into a major refactoring of how scripts ran in `ch` before I spent too much time on it and gave up. @rhuanjl , if you are interested, I think its a great feature to have; otherwise, I can look into it.

**Comment by rhuanjl** [comment]
@jackhorton I've got 2 hours free now will see if I can do it in that time, if not I'll leave it to you.

**Comment by MikeHolman** [comment]
@rhuanjl I think your implementation looks fine, and don't think it needs TTD support.

**Comment by rwaldron** [comment]
@rhuanjl 

> Depending on what you're testing you may also need #5237 resolving.

All of Test262's module tests are required to have _ImportSpecifier_ written as: `'./foo.js'` ([INTERPRETING.md#modules](https://github.com/tc39/test262/blob/master/INTERPRETING.md#modules)), but any disparity in host path resolution can be handled by `eshost` as needed.

**Comment by jackhorton** [comment]
I think that looks fine -- I had run into issues with the host path issues that #5237 and wasn't satisfied with not fixing both when I looked into it, but for this more urgent case, 5237 can stay a separate issue.

**Comment by rhuanjl** [comment]
@rwaldron The issue with #5237 would be if you had a test that involved multiple files in different folders as ch will evaluate each module specifier as being relevant to the CWD as of when ch was launched - ignoring the path of the parent module. If all your files for each test are in the same folder it will be fine (as long as ch is run with that as the CWD).

**Comment by rwaldron** [comment]
@rhuanjl thanks for the clarification—since Test262 requires that module code tests keep their dependencies in the same directory, this will be no problem. I appreciate the super quick feedback and feature implementation turnaround :)

**Comment by dilijev** [comment]
See Microsoft/ChakraCore#3284

**Comment by rwaldron** [comment]
Just want to say thanks for the quick work and excellent support here. ❤️
